vm/compiler/src/coinbase_puzzle/hash.rs

```diff
@@ -0,0 +1,81 @@
+// Copyright (C) 2019-2022 Aleo Systems Inc.
+// This file is part of the snarkVM library.
+
+// The snarkVM library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The snarkVM library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with the snarkVM library. If not, see <https://www.gnu.org/licenses/>.
+
+use console::prelude::{bail, cfg_into_iter, ensure, Result, Zero};
+use snarkvm_algorithms::{fft::DensePolynomial, polycommit::kzg10::KZGCommitment};
+use snarkvm_curves::PairingEngine;
+use snarkvm_fields::PrimeField;
+use snarkvm_utilities::CanonicalSerialize;
+
+use blake2::Digest;
+
+#[cfg(feature = "parallel")]
+use rayon::prelude::*;
+
+pub fn hash_to_coefficients<F: PrimeField>(input: &[u8], num_coefficients: u32) -> Result<Vec<F>> {
```

Comment:
```suggestion
pub fn hash_to_coefficients<F: PrimeField>(input: &[u8], num_coefficients: u32) -> Vec<F> {
```
it doesn't seem like the `Result` is needed here - all the operations are infallible